### ISSUE TITLE
What do we think about announcements?

### ISSUE THREAD

**Issue by ArtOfCode-** [Charcoal-SE/metasmoke#187]
Ref: https://chat.stackexchange.com/transcript/message/37871003#37871003

TL;DR: we could build some sort of "announcements" (for want of a better name) thing into metasmoke, and use it to help make people aware of current spam affairs.

The recent persistent holocaust denier troll kicked this thought off - if we had a thing on metasmoke that let people know about them, we'd get better communication, more accurate feedback, and potentially more co-ordinated efforts to stop the abuse.

I'm thinking pretty much just a free-form text field to allow (probably core) users to post an announcement, which would be displayed on the metasmoke homepage to logged-in users, either for *n* days or until a preset expiry date. Could also be emailed out to users we have an email for, if we wanted.

Thoughts? Shouldn't be too difficult to dev.

**Comment by yvettec** [comment]
This is a really good idea. I think we'd need a pinned post to remind us to go to the announcements page for a while, as I'm not sure how routinely people go to the MS homepage. 
Or
If people write an announcement, they can put a comment in chat and ping for  a RO to pin it.
As for emails, I am happy to be emailed, and that's something people can easily opt in for or not.

**Comment by AWegnerGitHub** [comment]
I don't think a dedicated announcements page is needed. Instead, add an announcements div to most/all pages of Metasmoke. Do we have any kind of usage statistics for metasmoke? Do we know what page(s) users see most often?

I am thinking something like this at the top of the appropriate pages. 

![Status](https://i.imgur.com/C55tY0Q.png)

This way, important announcements are seen whenever someone is on metasmoke and we don't need to train users to visit just to see the announcements.

> Could also be emailed out to users we have an email for, if we wanted.

If I can turn those off for myself, fine. If not, I'm going to be unhappy with someone.

**Comment by ArtOfCode-** [comment]
I don't think we have detailed usage stats, @AWegnerGitHub, and I was thinking along the lines of your extra div rather than a page. And yeah, emails would be opt-in.

**Comment by yvettec** [comment]
oh yeh, I meant a div. I agree not a page.

**Comment by Undo1** [comment]
I could come up with numbers for which pages are visited most. Not sure how this'd be better than pinning things in CHQ, though. Might help us clean up the sidebar, if that's a consideration?

**Comment by ArtOfCode-** [comment]
That's sorta what I was thinking, @Undo1. There's only a certain number of items we can sustainably have on the pinboard, and if we have more things than that to announce then we're a bit stuck.

**Comment by angussidney** [comment]
Remember @Undo1, a lot of people don't read the sidebar. 

Also, having these annoucements on MS/available by email allows moderators etc who don't visit CHQ to keep up to date with affairs.

**Comment by ArtOfCode-** [comment]
That's actually a decent point - if we're posting about all things spam, a bunch of moderators might be interested in seeing those too. Might be worth looking at telling TL.

**Comment by superplane39** [comment]
But we should make it so that once you've dismissed an announcement, don't keep showing that *same* announcement. Once you've clicked 'Hide announcements', it should hide that announcement until a new one is put up, instead of having to close it again every time you visit a new page/refresh on metasmoke.

**Comment by NobodyNada** [comment]
@superplane39 I'd like that too, but, as Art explained in another issue, it'd be a bit more difficult to implement since it would require creating a join table between users and announcements.

I agree that it'd be nice to have, though, and if I have time I might try implementing it myself (but I'm not promising anything).

**Comment by ArtOfCode-** [comment]
Implemented now, including sticky show/hide of already-read announcements.
